Sources/PostgresNIO/New/Connection State Machine/ExtendedQueryStateMachine.swift

```diff
@@ -4,7 +4,7 @@ struct ExtendedQueryStateMachine {
     
     private enum State {
         case initialized(ExtendedQueryContext)
-        case parseDescribeBindExecuteSyncSent(ExtendedQueryContext)
+        case messagesSent(ExtendedQueryContext)
```

Comment:
Is the name `messageSent` intentionally generic? When I read it my first reaction was "What message? Do we have different kind of messages flying around?"

Reply:
yes. it depends on what the internal QueryContext says. 

If we run an unnamed query, we sent `parse, describe, bind, execute & sync`.
If we prepare a statement, we sent `parse, describe & sync`.
If we execute a statement, we sent `bind, execute & sync`.